pkg/envoy/admin.go

```diff
@@ -0,0 +1,119 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package envoy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	envoyAdmin "github.com/envoyproxy/go-control-plane/envoy/admin/v2alpha"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/proto"
+)
+
+// Shutdown initiates a graceful shutdown of Envoy.
+func Shutdown(adminPort uint32) error {
```

Comment:
FYI this only does graceful shutdown of HTTP connections IIRC.

Reply:
Yeah that probably makes sense ... is your suggestion to update documentation?

Reply:
I didn't really have a suggestion. Maybe just add a note as a comment?

Reply:
Taking a quick glance at the envoy docs, I didn't see anything regarding how this works. Do you have a link handy? My concern is adding docs here that aren't quite right.

Reply:
This is the reason we don't use it for graceful shutdown and instead swap to config with no listener and then issue a terminate. See https://github.com/istio/istio/issues/7136#issuecomment-456441540 for some discussion.